cmd/karmaExecuteTests.go

```diff
@@ -11,12 +11,10 @@ import (
 )
 
 func karmaExecuteTests(myKarmaExecuteTestsOptions karmaExecuteTestsOptions) error {
-	karmaLogger := log.Logger().WithField("stepName", "karmaExecuteTests")
-
 	c := command.Command{}
 	// reroute command output to loging framework
-	c.Stdout = karmaLogger.Writer()
-	c.Stderr = karmaLogger.WriterLevel(logrus.ErrorLevel)
+	c.Stdout = log.Logger().Writer()
+	c.Stderr = log.Logger().WriterLevel(logrus.ErrorLevel)
```

Comment:
Maybe this is a misunderstanding here.

The coding just captures everything which is written to Stdout/StdErr by the command and brings this into the logging infrastructure.

The tooling called here seems to behave in a way that errors are written to Stderr while e.g. INFO is written to Stdout.
Assuming this is the case, then code is fine from my POV.
If the tooling behaves in a way that also non-error messages are written to Stderr then log level should not be ERROR as suggested above.

Slightly different topic but related, Maven for example logs everything to Stdout (see https://maven.apache.org/maven-logging.html) and thus behaves again differently.

Thus we should use an adequate approach per executable for log redirection of Stdout/Stderr since different tools behave differently.

Reply:
> The coding just captures everything which is written to Stdout/StdErr by the command and brings this into the logging infrastructure.

I know, I know. No misunderstanding.

>The tooling called here seems to behave in a way that errors are written to Stderr while e.g. INFO is written to Stdout.

That is a crazy behaviour, I know about such cases. Example already mentioned listed: maven. But it is a cool way to argue with bad examples?

I think we might have steps returning some reasonable payload through `stdout`, at least we should not  close that door here by using stdout for log messages (also not in cases where underlying tools are doing that). And I think the lib should behave consistently for all the steps: No having some steps mis-using `stdout` for logging while other deliver payload.

E.g. for transportRequestCreate it would be a convenient way to return the ID of the created transport request like this, maybe we agree also on returning some config.yml through stdout. We should not close that door that early. IMO using `stdout` for logging here would be a bad decision. 

Hence I would like so suggest again not to use `stdout` for logging, but keeping that stream clear for payload delivered through that.

Reply:
Bringing Stdout and Stderr to the log is not a general implementation here but step-specific.

Since the tool called writes information like that it needs to be captured to bring it to the log.

For other tools doing it differently one for sure need to do it as the tool requires ...

@CCFenner could you please check if using ERROR level in this case really makes sense. If there are non-ERROR cases also written to Stdout I suggest to change it accordingly ...

Reply:
It seems that only the NPM errors are logged to `stderr`, Karma errors are logged to `stdout`. So I would agree to log everything from `stdout` & `stderr` with severity `INFO`. 

```
INFO[0005] 04 11 2019 11:36:22.434:INFO [WebDriver]: Killed Karma test.  stepName=karmaExecuteTests
INFO[0005] 04 11 2019 11:36:22.434:ERROR [launcher]: Cannot start chrome via Remote WebDriver  stepName=karmaExecuteTests
INFO[0005] 04 11 2019 11:36:22.434:ERROR [launcher]: chrome via Remote WebDriver stdout:   stepName=karmaExecuteTests
INFO[0005] 04 11 2019 11:36:22.434:ERROR [launcher]: chrome via Remote WebDriver stderr:   stepName=karmaExecuteTests
INFO[0005] 04 11 2019 11:36:22.434:ERROR [launcher]: chrome via Remote WebDriver failed 2 times (cannot start). Giving up.  stepName=karmaExecuteTests
INFO[0005] [11:36:22] 'test:opa' errored after 314 ms    stepName=karmaExecuteTests
INFO[0005] [11:36:22] Error: 1                           stepName=karmaExecuteTests
ERRO[0005] npm ERR! code ELIFECYCLE                      stepName=karmaExecuteTests
ERRO[0005] npm ERR! errno 1                              stepName=karmaExecuteTests
```

Reply:
Not sure if that is a really cool idea, but that comes just to my mind:

Karma seems to use some logging framework. That means: what karma delivers is already somehow log-formatted. Why pushing this again to another log framework?

Why not redirecting everything from karma to `stderr`? With that `stdout` is kept clear (... that is really important) and the already log-formatted karma output is contained in ´stderr` so that we can read it if we need to do so.